e2e_test/udf/test.py

```diff
@@ -22,16 +24,25 @@ def gcd3(x: int, y: int, z: int) -> int:
     return gcd(gcd(x, y), z)
 
 
+@udf(input_types=['BINARY'], result_type='STRUCT<src_ip VARCHAR, dst_ip VARCHAR, src_port SMALLINT, dst_port SMALLINT>')
```

Comment:
🤔 Is it possible to use Python3 type hints?

Reply:
Do you mean to infer SQL type from function signatures?

Reply:
yep, just like how `@dataclass` do

Reply:
Interesting idea. It seems feasible.

I can think of one more benefit: a python function can be compatible with multiple SQL functions.
For example: `def gcd(x: int, y: int): int` can be used as SQL functions:
- `gcd(INT2, INT2) -> INT2`
- `gcd(INT4, INT4) -> INT4`
- `gcd(INT8, INT8) -> INT8`
- ...